src/components/SignUp/SignUp.js

```diff
@@ -0,0 +1,170 @@
+import React, { useState } from 'react';
+import SVG from '../../assets/login.svg';
+import {
+	Heading,
+	Flex,
+	VStack,
+	Image,
+	SimpleGrid,
+	GridItem,
+	FormControl,
+	FormLabel,
+	Input,
+	Text,
+	Button,
+} from '@chakra-ui/react';
+import { Link } from 'react-router-dom';
+import axios from 'axios';
+import * as Yup from 'yup';
+
+export default function SignUp() {
+	const [firstName, setFirstName] = useState('');
+	const [lastName, setLastName] = useState('');
+	const [email, setEmail] = useState('');
+	const [password, setPassword] = useState('');
+	const [repeatPassword, setRepeatPassword] = useState('');
+	const [error, setError] = useState(false); // if any input validation error occur
+
+	// input validation schema
+	const signupSchema = Yup.object().shape({
+		firstName: Yup.string().max(50).required(),
+		lastName: Yup.string().max(50).required(),
+		email: Yup.string().email().required(),
+		password: Yup.string().min(8).max(50).required(),
+		repeatPassword: Yup.string()
+			.oneOf([Yup.ref('password'), null])
+			.required(),
+	});
+
+	const handleClick = async (e) => {
+		e.preventDefault();
+		setError(false); // resetting the error messgae
+		const isValid = await signupSchema.isValid({
+			firstName: firstName,
+			lastName: lastName,
+			email: email,
+			password: password,
+			repeatPassword: repeatPassword,
+		});
+
+		if (isValid) {
+			const URI = 'http://localhost:5000/signup'; // URI is subject to change
```

Comment:
api path will be 'http://localhost:5000/api/auth/signup' on backend

Reply:
done